Enable WASM AOT microbenchmark run

Lxiamail opened the pull request.

Lxiamail committed:
bring up WASM AOT microbenchmarks run

Lxiamail committed:
bring up WASM AOT microbenchmarks run

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
adding diagnostice log for testing

Lxiamail committed:
commend out some jobs while testing to save lab run resource

Lxiamail committed:
commend out some jobs while testing to save lab run resource

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Add WASM AOT in not scheduled run

Lxiamail committed:
Only run selected jobs

Lxiamail committed:
add "ls -l" to print out permission before calling performance-setup.sh

Lxiamail committed:
add "ls -l" to print out permission before calling performance-setup.sh

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Add "--buildTimeout 3600" to benchmarkdotnet to extend default timeout time. Adjust -aotcompilermode format to "--aotcompilermode=wasm".

Lxiamail committed:
Add "--buildTimeout 3600" to benchmarkdotnet to extend default timeout time. Adjust -aotcompilermode format to "--aotcompilermode=wasm".

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
resolve merge conflict

Lxiamail committed:
resolve merge conflict

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Correct runtimeSrcDir value

Lxiamail committed:
Correct --runtimeSrcDir location to $HELIX_CORRELATION_PAYLOAD

Lxiamail committed:
Correct --runtimeSrcDir location to $HELIX_CORRELATION_PAYLOAD

Lxiamail committed:
merge conflict

Lxiamail committed:
Remove not used parameters in WASM AOT mode and  add temperary diagnostic info

Lxiamail committed:
Remove not used parameters in WASM AOT mode and  add temperary diagnostic info

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
make \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm to be the root directory of runtime repo source

Lxiamail committed:
Only copy runtime repo source directory to helix payload when it's wasm aot

Lxiamail committed:
Add diagnosic log for testing only

Lxiamail committed:
omite unneeded runtime repo file from helix payload

Lxiamail committed:
omite unneeded runtime repo file from helix payload

Lxiamail committed:
merge confict

Lxiamail committed:
use rsync instead of cp exclusion command since cp doesn't work with sub directory as expected.

Lxiamail committed:
Add temp diagnostic logging, should be removed before PR.

Lxiamail committed:
remove artifacts/obj from helix payload

Lxiamail committed:
remove artifacts/obj from helix payload

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
add "--keepfiles" to BDN for temperary diagnostic.Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
copy back helix job for diagnosis and silent rsync

Lxiamail committed:
For testing only, clone performance repor alicial/hikeMicro-benchmark private branch, which contains copying back helix job folder for diagnosis

Lxiamail committed:
enable rsync verbose for diagnosis

Lxiamail committed:
add --keepfiles for wasm intepreter as well for diagnosis

Lxiamail committed:
Remove temporary diagnostic code

Lxiamail committed:
recover all the jobs in the pipeline

Lxiamail committed:
recover all the jobs in the pipeline

Lxiamail committed:
merge config

Lxiamail committed:
merge config

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
copy wasm build drop to the location that aot build expects

Lxiamail committed:
using private perf branch to enable binlog for benchmarkdotnet for diagnosis

Lxiamail committed:
comment out none wasm runs temporarily for testing only

Lxiamail committed:
comment out none wasm runs temporarily for testing only

Lxiamail committed:
:merge conflict

Lxiamail committed:
add "--keepfiles" to keep diagnostic files

Lxiamail committed:
fixed $wasm_dotnet_path/artifacts/BrowserWasm/artifacts path

Lxiamail committed:
instead of copy, move $wasm_dotnet_path/artifacts/BrowserWasm/artifacts/* to save space

Lxiamail committed:
benchmarkdotnet using --cli \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/dotnet.sh instead

Lxiamail committed:
benchmarkdotnet using --cli \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/dotnet.sh instead

Lxiamail committed:
merge conflict

Lxiamail committed:
add executable permission

Lxiamail committed:
Enable other non run to compare

Lxiamail committed:
Enable other non run to compare

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
remove "--cli \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/dotnet.sh"

DrewScoggins commented on eng/testing/performance/performance-setup.sh:
```diff
@@ -264,7 +269,16 @@ if [[ "$wasm_runtime_loc" != "" ]]; then
+        # copy wasm build drop to the location that aot build expects
+        rsync -a --progress $wasm_dotnet_path/artifacts/BrowserWasm/artifacts/* $wasm_dotnet_path/artifacts
+        rm -r $wasm_dotnet_path/artifacts/BrowserWasm/artifacts
+        extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --wasmEngine /home/helixbot/.jsvu/$javascript_engine --aotcompilermode wasm --runtimeSrcDir \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm --buildTimeout 3600 --keepfiles" 
```
Make sure that --keep-files is removed

DrewScoggins commented on eng/pipelines/coreclr/templates/perf-job.yml:
```diff
@@ -54,9 +54,9 @@ jobs:
-      - ${{ format('build_{0}{1}_{2}_{3}_{4}', 'Browser', '', 'wasm', parameters.buildConfig, parameters.runtimeType) }}
-    - ${{ if eq(parameters.codeGenType, 'AOT')}}:
+    - ${{ if eq(parameters.runtimeType, 'wasm')}}:
+      - ${{ format('build_{0}{1}_{2}_{3}_{4}', 'Browser', '', 'wasm', parameters.buildConfig, parameters.runtimeType, parameters.codeGenType) }}
```
We never reference the codeGenType parameter, it is the 6th one, so we would need to see {5} in the string, but don't.

Lxiamail committed:
fixes per CR

Lxiamail committed:
copy EMSDK_PATH to expected location in helix payload

Lxiamail committed:
install emsdk

Lxiamail committed:
fix emsdk path

Lxiamail committed:
minor fix

Lxiamail committed:
removed more not used files to reduce helix payload size

Lxiamail committed:
remove more jobs for testing purpose

radekdoulik committed:
Fix the download path

radekdoulik committed:
Add BenchmarkDotNet sources to the payload

radekdoulik committed:
Fix __download__ path

Lxiamail committed:
set BenchmarkDotNetSources to use benchmarkdotnet built from source

Lxiamail committed:
use benchmarkdotnet private branch:alicial/tempwasmaot for BenchmarkDotNetSources

Lxiamail committed:
use benchmarkdonet private branch with BenchmarkDotNetSources option

radekdoulik committed:
Debug prints

radekdoulik committed:
Clone B.NET to different location

Fix rsync exclusion
Add more debug prints

radekdoulik committed:
Move B.NET back, clone upstream B.NET

To avoid:

    [INFO] ERROR(S):
    [INFO]   Option 'BenchmarkDotNetSources' is unknown.

radekdoulik committed:
Use my branch for performance repo

radekdoulik committed:
Revert "set BenchmarkDotNetSources to use benchmarkdotnet built from source"

This reverts commit bccfadde684922c0440d928224c8e7ec7c220fe9.

To avoid:

    [INFO] ERROR(S):
    [INFO]   Option 'BenchmarkDotNetSources' is unknown.

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into internal-alicial/WASMAOTMicrobenchmarks

radekdoulik committed:
Disable one more job I missed in merge

radekdoulik committed:
Switch back to nuget packages for BenchmarkDotNet

Lxiamail committed:
tempararily use pr-alicias-branch to test new fixes

Lxiamail committed:
merge change

vargaz committed:
[mono] Fix an uninitialized memory access in the ABCREM pass.

Detected by valgrind.

radekdoulik committed:
[mono] Fix invalid memory write

Fixes https://github.com/dotnet/runtime/issues/56526 and probably also
https://github.com/dotnet/runtime/issues/53546

Allocate enough memory when `SIZEOF_REGISTER == 4`, so that code like

    defs [ins->dreg + 1] = NULL;
    defs [ins->dreg + 2] = NULL;

doesn't write after allocated range.

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into internal-alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
clean up PR

Lxiamail committed:
Merge branch 'alicial/WASMAOTMicrobenchmarks' of https://dev.azure.com/dnceng/internal/_git/dotnet-runtime into alicial/WASMAOTMicrobenchmarks

Lxiamail committed:
add missing param

Lxiamail committed:
fix perf.yml and make wasm interpreter to be consistent with wasm AOT

Lxiamail committed:
comment out more job for fast testing

Lxiamail committed:
remove extra job

Lxiamail committed:
exclude copying __download__ dir instead of delete it.

Lxiamail committed:
add parameters.codeGenType

Lxiamail committed:
fix aot javascriptengine

Lxiamail committed:
correct livelibrariesbuildconfig from release to Release

Lxiamail committed:
install emsdk before copy $source_directory

Lxiamail committed:
add back all the temporarily commented out jobs in perf.yml

Lxiamail commented on eng/testing/performance/performance-setup.sh:
```diff
@@ -249,12 +259,18 @@ common_setup_arguments="--channel main --queue $queue --build-number $build_numb
     setup_arguments="--perf-hash $commit_sha $common_setup_arguments"
 else
-    git clone --branch main --depth 1 --quiet https://github.com/dotnet/performance $performance_directory
+    echo "Not running from perf repo"
```
line 263-269 should be revert back to the original code to clone main branch once PR https://github.com/dotnet/performance/pull/1902 is merged

Lxiamail committed:
clean up work around code

Lxiamail committed:
minor clean up

DrewScoggins commented on eng/testing/performance/performance-setup.sh:
```diff
@@ -228,7 +234,11 @@ if [[ "$mono_dotnet" != "" ]] && [[ "$monointerpreter" == "false" ]]; then
 if [[ "$wasm_runtime_loc" != "" ]]; then
-    configurations="CompilationMode=wasm RunKind=$kind"
+    if [[ "$wasmaot" == "true" ]]; then
+        configurations="CompilationMode=wasm AOT=true RunKind=$kind"
```
AOT -> MonoAOT to match the current configs we have.

Lxiamail replied:
Is MonoAOT different from wasm aot?

Lxiamail committed:
remove performance submodule

DrewScoggins commented on eng/testing/performance/performance-setup.sh:
```diff
@@ -254,8 +264,10 @@ if [[ "$run_from_perf_repo" = true ]]; then
 else
-    git clone --branch main --depth 1 --quiet https://github.com/dotnet/performance $performance_directory
-    
+    git clone --branch alicial/wasmaotmicro --depth 1 https://github.com/dotnet/performance.git $performance_directory
```
This needs to switch back to cloning from the performance repo.

DrewScoggins commented on eng/testing/performance/performance-setup.sh:
```diff
@@ -264,7 +276,21 @@ if [[ "$wasm_runtime_loc" != "" ]]; then
+    # install emsdk, $source_directory/src/mono/wasm/ has the nuget.config with require feed. EMSDK may be available in the payload in a different directory, should visit this install to avoid deplicated payload.
+    pushd $source_directory/src/mono/wasm/
+    make provision-wasm
+    EMSDK_PATH = $source_directory/src/mono/wasm/emsdk
```
This variable gets set, but it does not ever seem to get used anywhere. What is it for?

Lxiamail replied:
Yes, EMSDK_PATH is not used in our script file. It is set here is per document guidance at https://github.com/dotnet/runtime/blob/d953229d5429d2ffde833740dd481aab864d3e0c/src/mono/wasm/README.md

DrewScoggins reviewed (approved):
LGTM, apart from the few small things I mentioned in comments. ~~Also we need to remove the performance submodule~~

Lxiamail committed:
switch back to use performance repo main branch

Lxiamail committed:
add --quiet to git clone

Lxiamail committed:
Merge branch 'main' into alicial/WASMAOTMicrobenchmarks
